src/main/java/io/odpf/firehose/objectstorage/gcs/GCSObjectStorage.java

```diff
@@ -30,26 +32,23 @@ public GCSObjectStorage(GCSConfig gcsConfig) throws IOException {
     }
 
     @Override
-    public void store(String localPath) {
+    public void store(String localPath) throws IOException, ObjectStorageException {
         String objectName = gcsConfig.getLocalBasePath().relativize(Paths.get(localPath)).toString();
-        try {
-            byte[] content = Files.readAllBytes(Paths.get(localPath));
-            store(objectName, content);
-        } catch (IOException e) {
-            e.printStackTrace();
-            throw new ObjectStorageFailedException(e);
-        }
+        byte[] content = Files.readAllBytes(Paths.get(localPath));
+        store(objectName, content);
     }
 
     @Override
-    public void store(String objectName, byte[] content) throws IOException {
+    public void store(String objectName, byte[] content) throws ObjectStorageException {
         BlobId blobId = BlobId.of(gcsConfig.getGcsBucketName(), objectName);
         BlobInfo blobInfo = BlobInfo.newBuilder(blobId).build();
         try {
             storage.create(blobInfo, content);
             LOGGER.info("Created object in GCS " + blobInfo.getBucket() + "/" + blobInfo.getName());
         } catch (StorageException e) {
-            throw new IOException(e);
+            GCSErrorType gcsErrorType = GCSErrorType.valueOfCode(e.getCode());
```

Comment:
If we are not expecting this exception to be handled just throw it as a RuntimeException, it would make the error handling simpler

Reply:
We need to add metrics for these errors

Reply:
the metrics is already added here https://github.com/odpf/firehose/blob/8463aa3c1cf73419afa95e9e169aa69127b28986/src/main/java/io/odpf/firehose/sink/objectstorage/writer/remote/ObjectStorageChecker.java#L88